### DISCUSSION TITLE
Proposal: deprecate support for legacy type hinting functionality, i.e. [float] instead of list[float], {str: int} instead of dict[str,int], etc

### DISCUSSION THREAD

**Discussion by ptomecek** [Point72/csp#166] [Ideas]
**Is your feature request related to a problem? Please describe.**
Simplify the existing codebase and make performance improvements during graph parsing/type checking/building. These files would no longer be necessary:

- [container_type_normalizer.py](https://github.com/Point72/csp/blob/main/csp/impl/types/container_type_normalizer.py)
- [type_annotation_normalizer_transformer.py](https://github.com/Point72/csp/blob/main/csp/impl/types/type_annotation_normalizer_transformer.py)

**Describe the solution you'd like**
Add deprecation messages to use of the legacy container types, followed by complete removal down the road.

**Describe alternatives you've considered**
Leaving the functionality in. 

**Additional context**
Now that python type hints are standardized and ubiquitous, it doesn't make so much sense to support syntax that was developed long ago.

**Comment by AdamGlustein** [comment]
Since type hinting for generic containers (i.e. `list[int]`) only became supported by the Python parser in 3.9, I think we could only do this if we drop support for 3.8. We would at least need to maintain support for `typing.List` and other collections.

**Reply by timkpaine** [reply]
Python 3.8 is end of life very soon anyway https://devguide.python.org/versions/

**Reply by ptomecek** [reply]
To be clear, I think typing.List[float] should be supported on Python 3.8+ and list[float] on 3.9+. I don't think [float] should be supported at all, because it's not standard typing syntax. So the only things I think we should get rid of are the ones that aren't valid type hints for any version of python.

**Reply by timkpaine** [reply]
agreed on that, no reason to make up conventions when the existing mechanisms are sufficient
